### PR TITLE
DEV: Add class to header-topic-info widget

### PR DESCRIPTION
**Pull request by markvanlan**
There is an `.extra-info-wrapper` holding the `.extra-info` div. Right now, a `.two-rows` class gets added to `.extra-info` when viewing a topic with a category showing. If the topic is uncategorized, and suppress uncategorized is true, the `.two-rows` class is not present.

In **some plugins** we decorate this widget like so: `header-topic-info:after`. The problem here is that we need the `two-rows` class to be on the PARENT of `header-topic-info` so that we can use it as a css selector.

### CODE REVIEW

File: app/assets/javascripts/discourse/widgets/header-topic-info.js [hunk 1]
```diff
@@ -208,6 +208,14 @@ export default createWidget("header-topic-info", {
     );
   },
 
+  showCategory(topic) {
+    return (
+      topic.category &&
+      (!topic.category.get("isUncategorizedCategory") ||
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/widgets/header-topic-info.js:214]: do we need the `.get` here ?

Comment (markvanlan) [reply app/assets/javascripts/discourse/widgets/header-topic-info.js:214]: I can remove it!

File: app/assets/javascripts/discourse/widgets/header-topic-info.js [hunk 2]
```diff
@@ -112,11 +116,7 @@ export default createWidget("header-topic-info", {
     const category = topic.get("category");
 
     if (loaded || category) {
-      if (
-        category &&
-        (!category.get("isUncategorizedCategory") ||
-          !this.siteSettings.suppress_uncategorized_badge)
-      ) {
+      if (this.showCategory(topic)) {
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/widgets/header-topic-info.js:119]: I'm curious why don't we directly use category here ?

File: app/assets/javascripts/discourse/widgets/header-topic-info.js [hunk 3]
```diff
@@ -57,6 +57,10 @@ createWidget("topic-header-participant", {
 export default createWidget("header-topic-info", {
   tagName: "div.extra-info-wrapper",
 
+  buildClasses(attrs) {
+    return this.showCategory(attrs.topic) ? "two-rows" : "";
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/widgets/header-topic-info.js:61]: and `attrs.topic.category`  here ?

File: app/assets/javascripts/discourse/widgets/header-topic-info.js (hunk 2 above)
Comment (markvanlan) [reply app/assets/javascripts/discourse/widgets/header-topic-info.js:119]: I like the function name with topic as argument.

> Should we show the category for the given topic?

File: app/assets/javascripts/discourse/widgets/header-topic-info.js (hunk 3 above)
Comment (markvanlan) [reply app/assets/javascripts/discourse/widgets/header-topic-info.js:61]: Same as above. I can change it, but I do like how it reads.

File: app/assets/javascripts/discourse/widgets/header-topic-info.js (hunk 2 above)
Comment (jjaffeux) [reply app/assets/javascripts/discourse/widgets/header-topic-info.js:119]: well ok then :+1:

**Pull request merged**